core/drivers/imx/dcp/dcp.c

```diff
@@ -458,9 +441,9 @@ TEE_Result dcp_cipher_do_update(struct dcp_cipher_data *data,
 		data->payload_size += DCP_AES128_IV_SIZE;
 	}
 
-	desc->src_buffer = (uint8_t *)input.paddr;
-	desc->dest_buffer = (uint8_t *)output.paddr;
-	desc->payload = (uint8_t *)data->payload.paddr;
+	desc->src_buffer = (uint32_t)input.paddr;
```

Comment:
This casting shouldn't be needed. Please fix here and at the other newly introduced places.

Reply:
Ok